### PR TITLE
feat: introduce new error handling models

### PR DESCRIPTION
**Pull request by benjaminch**
Ticket: DEV-1004

### CODE REVIEW

File: src/errors/mod.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    message_safe: String,
+}
+
+#[derive(Debug)]
+pub enum LogLevel {
+    Debug,
+    Info,
+    Warning,
+    Error,
+    Critical,
```
Comment (erebe) [inline src/errors/mod.rs:19]: What the difference between Error and Critical ?

File: src/errors/mod.rs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    Upgrade,
+    Delete,
+}
+
+#[derive(Debug)]
+pub enum EnvironmentStep {
+    Build,
+    Deploy,
+    Update,
+    Delete,
```
Comment (erebe) [inline src/errors/mod.rs:41]: Pause ?

**Review by MacLikorne — APPROVED** [review]

File: src/errors/mod.rs (hunk 1 above)
Comment (benjaminch) [reply src/errors/mod.rs:19]: I would say Critical will fail the process instantly, but maybe Error should too, so I guess I can remove Critical.

File: src/errors/mod.rs (hunk 2 above)
Comment (benjaminch) [reply src/errors/mod.rs:41]: Sure ! Thanks, I wasn't exhaustive, this list will grow along migration :)

**Pull request merged**